### PR TITLE
Add `NewLineAfterSuperCall` Rule

### PR DESCRIPTION
**Pull request by hackertronix**
@pedrodanielcsantos [mentioned](https://github.com/Doist/Todoist-Android/pull/7354/files#r1706687245) a formatting issue on my PR

So I've created a new rule to ensure we don't forget about it 🙂

### CODE REVIEW

File: src/main/kotlin/com/doist/detekt/NewLineAfterSuperCall.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        Severity.Style,
+        "Ensure there is a new line after every call to a super() method.",
+        Debt.FIVE_MINS
+    )
+
+    override fun visitNamedFunction(function: KtNamedFunction) {
+        super.visitNamedFunction(function)
+        function.bodyBlockExpression?.statements?.forEachIndexed { index, statement ->
+            if (statement.text?.startsWith("super") == true) {
+                if (statement.nextSibling.text.contains("\n\n")) return
```
Comment (hackertronix) [inline src/main/kotlin/com/doist/detekt/NewLineAfterSuperCall.kt:24]: there's likely a cleaner way to do this, but I am not super on top of detekt APIs so for now sticking with this as it works!

File: src/main/kotlin/com/doist/detekt/NewLineAfterSuperCall.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        javaClass.simpleName,
+        Severity.Style,
+        "Ensure there is a new line after every call to a super() method.",
+        Debt.FIVE_MINS
+    )
+
+    override fun visitNamedFunction(function: KtNamedFunction) {
+        super.visitNamedFunction(function)
+        function.bodyBlockExpression?.statements?.forEachIndexed { index, statement ->
+            if (statement.text?.startsWith("super") == true) {
```
Comment (pedrodanielcsantos) [inline src/main/kotlin/com/doist/detekt/NewLineAfterSuperCall.kt:23]: Should we check here for `super.` (i.e. with the dot at the end), just for extra safety ?

File: src/main/kotlin/com/doist/detekt/NewLineAfterSuperCall.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    override val issue = Issue(
+        javaClass.simpleName,
+        Severity.Style,
+        "Ensure there is a new line after every call to a super() method.",
+        Debt.FIVE_MINS
+    )
+
+    override fun visitNamedFunction(function: KtNamedFunction) {
+        super.visitNamedFunction(function)
+        function.bodyBlockExpression?.statements?.forEachIndexed { index, statement ->
```
Comment (pedrodanielcsantos) [inline src/main/kotlin/com/doist/detekt/NewLineAfterSuperCall.kt:22]: A bit ironic but ... Can we add an empty line here, please? 😇😅

File: src/test/kotlin/com/doist/detekt/NewLineAfterSuperCallTest.kt [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+
+import io.gitlab.arturbosch.detekt.rules.KotlinCoreEnvironmentTest
+import io.gitlab.arturbosch.detekt.test.lint
+import org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+
+@KotlinCoreEnvironmentTest
+internal class NewLineAfterSuperCallTest(private val env: KotlinCoreEnvironment) {
```
Comment (pedrodanielcsantos) [inline src/test/kotlin/com/doist/detekt/NewLineAfterSuperCallTest.kt:11]: 👏

**Review by pedrodanielcsantos — APPROVED** [review]
Thanks a lot for the update @hackertronix ! 🚀

Left a couple of small suggestions, good to go after checking them. 

To make it the whole package, you can publish a new version and update Todoist afterwards. 🙏

File: src/main/kotlin/com/doist/detekt/NewLineAfterSuperCall.kt (hunk 2 above)
Comment (hackertronix) [reply src/main/kotlin/com/doist/detekt/NewLineAfterSuperCall.kt:23]: good catch

File: src/main/kotlin/com/doist/detekt/NewLineAfterSuperCall.kt (hunk 3 above)
Comment (hackertronix) [reply src/main/kotlin/com/doist/detekt/NewLineAfterSuperCall.kt:22]: LOL

**Pull request merged**